<a href="https://colab.research.google.com/github/PurvanshBarodia/price_predictor/blob/main/Purv_House_Price_Prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**What we are going to do?**

We wil try to train the model to predict house price and then see how much better a model can pridict on unseen data.

Which data we will use?

In [2]:
# Upload downloaded csv file
from google.colab import files
uploaded = files.upload()

Saving housing.csv to housing.csv


In [4]:
# What is pandas and how it is useful?
import pandas as pd

In [6]:
df = pd.read_csv('housing.csv')

Let's explore the dataset.

In [7]:
# Dataset contain 20640 rows and 10 cols.
df.shape 

(20640, 10)

In [8]:
# See first five rows of dataset.
df.head()

longitude  latitude  ...  median_house_value  ocean_proximity
0    -122.23     37.88  ...            452600.0         NEAR BAY
1    -122.22     37.86  ...            358500.0         NEAR BAY
2    -122.24     37.85  ...            352100.0         NEAR BAY
3    -122.25     37.85  ...            341300.0         NEAR BAY
4    -122.25     37.85  ...            342200.0         NEAR BAY

[5 rows x 10 columns]

In [9]:
# Let's understand Which are the columns and what it represents?
df.columns

Index(['longitude', 'latitude', 'housing_median_age', 'total_rooms',
       'total_bedrooms', 'population', 'households', 'median_income',
       'median_house_value', 'ocean_proximity'],
      dtype='object')

There are 10 cols:

1. longitude: A measure of how far west a house is; a higher value is farther west

2. latitude: A measure of how far north a house is; a higher value is farther north

3. housingMedianAge: Median age of a house within a block; a lower number is a newer building

4. totalRooms: Total number of rooms within a block

5. totalBedrooms: Total number of bedrooms within a block

6. population: Total number of people residing within a block

7. households: Total number of households, a group of people residing within a home unit, for a block

8. medianIncome: Median income for households within a block of houses (measured in tens of thousands of US Dollars)

9. medianHouseValue: Median house value for households within a block (measured in US Dollars)

10. oceanProximity: Location of the house w.r.t ocean/sea

 Above data description tells that rather than looking for a single house we are going to look at all houses whithin a block to predict house price.

In [16]:
# Let's take statistical overview of each numarical cols
df.describe()

longitude      latitude  ...         incom          price
count  20640.000000  20640.000000  ...  20640.000000   20640.000000
mean    -119.569704     35.631861  ...      3.870671  206855.816909
std        2.003532      2.135952  ...      1.899822  115395.615874
min     -124.350000     32.540000  ...      0.499900   14999.000000
25%     -121.800000     33.930000  ...      2.563400  119600.000000
50%     -118.490000     34.260000  ...      3.534800  179700.000000
75%     -118.010000     37.710000  ...      4.743250  264725.000000
max     -114.310000     41.950000  ...     15.000100  500001.000000

[8 rows x 9 columns]

In [ ]:
# optional
# what standard deviation suggest? high std means what?

In [11]:
# Now, rename the column names as some col names are....
new_columns = ['longitude','latitude','house_age','rooms','bedrooms','poputation','households','incom','price','ocen_proximity']
df.columns = new_columns

In [12]:
# Let's apply data cleaning, so that we can use our data to train the model

In [13]:
# Check howmany null values are there in each columns.

In [15]:
df.isnull().sum()

longitude           0
latitude            0
house_age           0
rooms               0
bedrooms          207
poputation          0
households          0
incom               0
price               0
ocen_proximity      0
dtype: int64

Here we have 207 missing values for bedrooms, we can:
1. Drop the entire row wich have missing value.

2. We can fill those null vlaues with appropriate statistic (mean, median or mode)